core/src/main/java/org/apache/calcite/sql2rel/RelFieldTrimmer.java

```diff
@@ -697,14 +697,20 @@ public TrimResult trimFields(
 
   /**
    * Variant of {@link #trimFields(RelNode, ImmutableBitSet, Set)} for
-   * {@link org.apache.calcite.rel.core.SetOp} (including UNION and UNION ALL).
+   * {@link org.apache.calcite.rel.core.Union}.
```

Comment:
I think a doc here to explain why not trim other SetOp types might be helpful for others.

Reply:
Hmm, this is a breaking API change.

Reply:
@hsyuan 
How about we keep the API interface as before and only do the trimming when type of `SetOp` is union ?
How do you think ?
But it seems weird since Calcite really means to only trim Union and it's incorrect by semantics on other operators.

Reply:
Sure, updated ~
Regarding the sanity check, I just check and skip the trimming logic when it's not `UNION ALL`.